File: 2015/day4/part1.js
const crypto = require("crypto");
let code = "";
let str = 0;
while (code.substring(0, 5) != "00000") {
  str++;
  console.log(str);
  code = crypto.createHash("md5").update(`bgvyzdsv${str}`).digest("hex");
}

console.log(str);

File: 2015/day4/python/part1.py
import hashlib

suffix = 0
code = 'bgvyzdsv'
# result = hashlib.md5(b"Hello MD5").hexdigest()
hash = ""

while hash[0:5] != "00000":
    suffix +=1
    hash = hashlib.md5((code + str(suffix)).encode('utf-8')).hexdigest()

print(suffix)